api/src/main/java/org/openmrs/module/eptsreports/reporting/library/datasets/TxRetDataset.java

```diff
@@ -44,18 +45,21 @@
   @Autowired private TxRetDimensionCohort txRetDimensionCohort;
```

Comment:
There is need to start injecting this dependencies through a constructor which is the recommended standard. I started cleaning those up in https://jembiprojects.jira.com/browse/MEPTS-185

Reply:
are we tending to migrating from autowiring dependencies, can i know more about this decision...

Reply:
I understood that we'll try use the constructor injection method when Autowired does not work so basically instead of using `Context.get...`

What is the benefit of replacing Autowired with injection via the constructor?

Reply:
Gentle men by inspecting code in IntelliJ I found recommendations about those. see the image attached, and that what informed me to have those changed in MEPTS-185, though it does NOT stop us from working, but they are flagged as warnings in my IDE.
![injecting](https://user-images.githubusercontent.com/1763308/59087459-511dd300-890d-11e9-8da0-7bf182b5037e.png)

Reply:
Look at the field injection, so I ended up doing something like this https://github.com/esaude/openmrs-module-eptsreports/blob/MEPTS-185_Refactor_all_obs_queries/api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TxCurrCohortQueries.java#L42-L56, but that can be debatable

Reply:
Ok cool, it is still nice and neat and since it seems to be recommended by the Spring Team I guess there is a reason they prefer this approach :+1: